src/software/amazon/ion/impl/PrivateUtils.java

```diff
@@ -228,10 +162,11 @@ public static SymbolToken newSymbolToken(SymbolTable symtab,
                                              String text)
     {
         // TODO amzn/ion-java#21 symtab should not be null
-        if (text == null || text.length() == 0)
+        if (text == null)
         {
             throw new EmptySymbolException();
```

Comment:
I agree this would be better as `NPE` or `IAE`. However I was avoiding removing `EmptySymbolException` whenever possible as it breaks binary compatibility for some interfaces, since `EmptySymbolException` is a checked exception.

I'm pro removing it as a whole if we consider are OK with this kind of compatibility break